src/components/NavBar/NavBar.component.jsx

```diff
@@ -1,17 +1,39 @@
-import React from 'react';
-import { NavBar as Navigation, SearchField, CheckBox, CheckBoxLabel, CheckBoxWrapper, LoginButton } from './styled';
+import React, { useState } from 'react';
+import { NavBar as Navigation, SearchFieldWrapper, SearchField, CheckBox, CheckBoxLabel, CheckBoxWrapper, LoginButton, RightDiv } from './styled';
+
+const NavBar = ({ onVideosChanged }) => {
+    const [searchValue, setSearchValue] = useState('elon musk on TED Talks');
+    const [darkMode, setDarkMode] = useState(true);
+    const baseApiUrl = 'https://youtube.googleapis.com/youtube/v3/search';
+
+    async function searchVideos(keystroke) {
+        if (keystroke === 'Enter') {
+            let response = await fetch(`${baseApiUrl}?part=snippet&maxResults=25&q=${searchValue}&key=${process.env.REACT_APP_API_KEY}`);
+
+            if (response.ok) { // if HTTP-status is 200-299
+                let json = await response.json();
+                onVideosChanged(json.items);
+            } else {
+                alert(`"HTTP-Error: " + ${response.status}\nSearch: Failed\nProbably because API_KEY is not set in your env`);
+            }
+        }
+    }
```

Comment:
Some of this code is duplicated in other components, when you fetch data you have something like:
```jsx
const response = await fetch(url);
if (response.ok) {
  // get json change state
} else {
  // handle error
}
```
The way you build the URL and process the response could be abstracted and create a custom hook, this way you remove duplicated code and improve the code quality overall